index.js

```diff
@@ -1,61 +1,111 @@
-function highlightQuery (query, errors) {
-  var locations = errors.map(function (e) { return e.locations })
-    .reduce(function (a, b) {
-      return a.concat(b)
-    }, [])
+/* global fetch, Headers */
+require('isomorphic-fetch')
 
-  var queryHighlight = ''
+function Client (options) {
+  if (!options.url) throw new Error('Missing url parameter')
 
-  query.split('\n').forEach(function (row, index) {
-    var line = index + 1
-    var lineErrors = locations.filter(function (loc) { return loc.line === line })
+  this.options = options
+  this.url = options.url
+  // A stack of registered listeners
+  this.listeners = []
+}
+
+// to reduce file size
+var proto = Client.prototype
+
+/**
+ * Send a query and get a Promise
+ * @param   {String}   query
+ * @param   {Object}   variables
+ * @param   {Function} beforeRequest hook
+ * @returns {Promise}
+ */
+proto.query = function (query, variables, beforeRequest) {
+  var self = this
 
-    queryHighlight += row + '\n'
+  var headers = new Headers()
+  headers.set('content-type', 'application/json')
 
-    if (lineErrors.length) {
-      var errorHighlight = []
+  var req = self.options.request || {}
```

Comment:
I had to switch back to plain req object since there were many problems with Request. For example, I couldn't pass headers to it when initializing the client because they could only be modified after the Request object is initialized. Also, I couldn't pass "credentials" to it, because of [a bug](https://github.com/matthew-andrews/isomorphic-fetch/issues/102) in `isomorphic-fetch`. There were some other issues like a difference in behaviour between browser and Node, so it is just not the way to go